accounts-db/src/tiered_storage/footer.rs

```diff
@@ -204,8 +207,9 @@ impl TieredStorageFooter {
     }
 
     pub fn write_footer_block(&self, file: &TieredStorageFile) -> TieredStorageResult<()> {
-        file.write_type(self)?;
-        file.write_type(&TieredStorageMagicNumber::default())?;
+        // SAFETY: The footer does not contain any uninitialized bytes.
+        unsafe { file.write_type(self)? };
+        file.write_pod(&TieredStorageMagicNumber::default())?;
```

Comment:
Aghh!  I think I started understanding the logic here.

Btw, do we want to add an `assert!` here?  Something like assert whether `Self::sanitize(&footer)` is an ok?